hw/mcu/stm/stm32_common/src/hal_os_tick.c

```diff
@@ -76,3 +79,111 @@ os_tick_init(uint32_t os_ticks_per_sec, int prio)
     DBGMCU->CR |= (DBGMCU_CR_DBG_STOP | DBGMCU_CR_DBG_STANDBY);
 #endif
 }
+
```

Comment:
IMHO such a large block of STM32F1 specific code should not be the stm32_common files. Should be in the specific MCU directory instead... especially if we then have tickless code for the STM32L1 (as per my PR), STM32L0/L4, etc etc as each one requires a different approach.